Conceptpower+Spring/src/main/java/edu/asu/conceptpower/rest/ConceptSearch.java

```diff
@@ -118,6 +118,11 @@ private void initBinder(WebDataBinder binder) {
             } else if (SearchParamters.NUMBER_OF_RECORDS_PER_PAGE.equalsIgnoreCase(field.getName())) {
                 numberOfRecordsPerPage = field.get(conceptSearchParameters) != null
                         ? (Integer) field.get(conceptSearchParameters) : numberOfRecordsPerPage;
+            } else if (field.get(conceptSearchParameters) != null
+                    && (SearchParamters.EQUAL_TO.equalsIgnoreCase(field.getName())
+                            || SearchParamters.SIMILAR_TO.equalsIgnoreCase(field.getName()))) {
+                searchFields.put(field.getName().trim(),
```

Comment:
instead of introducing a new if for each property that needs some preprocessing, try using the propertyDescriptor:
https://docs.oracle.com/javase/7/docs/api/java/beans/PropertyDescriptor.html